sdks/java/io/mongodb/src/main/java/org/apache/beam/sdk/io/mongodb/MongoDbIO.java

```diff
@@ -306,30 +312,44 @@ public long getEstimatedSizeBytes(PipelineOptions pipelineOptions) {
           // the range from the beginning up to this split
           rangeFilter = String.format("{ $and: [ {\"_id\":{$lte:ObjectId(\"%s\")}}",
               splitKey);
+          filters.add(formatFilter(rangeFilter, additionalFilter));
         } else if (i == splitKeys.size() - 1) {
           // this is the last split in the list, the filter defines
```

Comment:
The comment should mention that we have two splits when dealing with the last element.